src/intl/context.js

```diff
@@ -1,16 +1,22 @@
-import FTLRuntimeParser from '../ftl/entries/parser';
 import { format } from './resolver';
 
+const { classes: Cc, interfaces: Ci, utils: Cu } = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "L20nParser",
+                                  "resource://gre/modules/L20nParser.jsm");
+
 export class MessageContext {
-  constructor(lang, { functions = {} } = {}) {
-    this.lang = lang;
+  constructor(locales, { functions = {}, args = {} } = {}) {
+    this.locale = Array.isArray(locales) ? locales[0] : locales;;
```

Comment:
It's different. `DateTimeFormat` takes a list of locales because it negotiates the best language given the installed ICU dataset.  `MessageContext` doesn't take its translations from the ICU;  they're passed to it via `.addMessage()`.  It isn't possible for `MessageContext` to end up being in a different language than the one supplied.

And yes, we want to standardize it.